dbcon/dmlpackageproc/commandpackageprocessor.cpp

```diff
@@ -374,6 +374,9 @@ DMLPackageProcessor::DMLResult CommandPackageProcessor::processPackage(
 
           if (weRc == 0)
           {
+            // MCOL-5021
+            fDbrm->addToLBIDList(fSessionID, lbidList);
```

Comment:
This is ad-hoc addition IMHO. Why does lbidList have no LBID's related with the fSessionID added in line 126, namely: fDbrm->getUncommittedExtentLBIDs()? How does it happen that aux column relates to fSessionID and not to txnid.id used in the line 126?

Reply:
No this is not an ad-hoc change. The DELETE code has changed now and we only perform version buffering on the aux column and not on the actual column. Due to this, `fDbrm->getUncommittedExtentLBIDs()` only returns the lbids for the aux column. Consider the following scenario:

```sql
create table (a int)engine=columnstore;
insert into t1 values (1), (2), (3);
select a from t1; -- a's extent min/max will be 1/3 respectively.
set autocommit=off;
delete from t1 where a in (1, 3);
select a from t1; -- a's extent min/max will be set to 2/2 now.
rollback;
select a from t1; -- this returns the correct 3 rows but the extent min/max for column a is incorrectly set to 2/2.
```

Without the above `fDbrm->addToLBIDList()` call, after the rollback, the extent min/max value for column a will not be invalidated and will still have the values 2 and 2 both for the min and the max, even though the column values are (1, 2, 3). This will cause the queries such as this one returning incorrect results after the rollback: `select a from t1 where a < 2;`

This was a test001 regression which is fixed with the `addToLBIDList()` change.

`fSessionID` passed in the call to `fDbrm->addToLBIDList()` is merely to get the singleton syscat object:

```
void DBRM::addToLBIDList(uint32_t sessionID, vector<LBID_t>& lbidList)                                                                       
{                                                                                                                                            
  boost::shared_ptr<execplan::CalpontSystemCatalog> systemCatalogPtr =                                                                       
    execplan::CalpontSystemCatalog::makeCalpontSystemCatalog(sessionID);                                                                     
```